### PR TITLE
samples: boards: nordic: system_off: Add comparator wakeup

### PR DESCRIPTION
**Pull request by nordic-babu**
Extend sytem_off sample to use analog comparator as wake up source.

### CODE REVIEW

File: samples/boards/nordic/system_off/Kconfig [hunk 1]
```diff
@@ -21,9 +21,20 @@ config APP_USE_RETAINED_MEM
 
 endchoice
 
+config GPIO_WAKEUP_ENABLE
+	bool "Use button to wake up device from system off"
+	default y
+help
```
Comment (masz-nordic) [inline samples/boards/nordic/system_off/Kconfig:27]: Alignment

File: samples/boards/nordic/system_off/src/main.c [hunk 2]
```diff
@@ -74,6 +80,10 @@ int main(void)
 
 	printf("Entering system off; press sw0 to restart\n");
 #endif
+#if defined(CONFIG_LPCOMP_WAKEUP_ENABLE)
+	comparator_set_trigger(comp_dev, COMPARATOR_TRIGGER_BOTH_EDGES);
```
Comment (bjarki-andreasen) [inline samples/boards/nordic/system_off/src/main.c:84]: the comparator could already be pending here, it should be cleared before going to sleep. Call `(void)comparator_trigger_is_pending()` to clear the pending status.

**Review by bjarki-andreasen — CHANGES_REQUESTED** [review]

File: samples/boards/nordic/system_off/src/main.c (hunk 2 above)
Comment (nordic-babu) [reply samples/boards/nordic/system_off/src/main.c:84]: done

**Comment by nordic-babu** [comment]
need to be adjusted when https://github.com/zephyrproject-rtos/zephyr/pull/82262 is merged

**Review by bjarki-andreasen — APPROVED** [review]
nice :)

**Review by kartben — APPROVED** [review]

**Pull request merged**